### PR TITLE
Add DPS310 to Matek boards F722,F722-SE,F722-HD and F722-PX

### PR DESCRIPTION
**Pull request by JulioCesarMatias**
https://github.com/emuflight/EmuFlight/issues/551

### CODE REVIEW

File: src/main/sensors/barometer.c [hunk 1]
```diff
@@ -40,6 +40,7 @@
 #include "drivers/barometer/barometer_fake.h"
 #include "drivers/barometer/barometer_ms5611.h"
 #include "drivers/barometer/barometer_lps.h"
+#include "drivers/barometer/barometer_dps310.h"
```
Comment (JulioCesarMatias) [inline src/main/sensors/barometer.c:43]: the extension is included but returning an error

Comment (JulioCesarMatias) [inline src/main/sensors/barometer.c:43]: the extension is included but returning an error

Comment (Quick-Flash) [reply src/main/sensors/barometer.c:43]: you need to add some code to the makefile to ensure that the barometer_dps310.c file is being compiled.

Comment (JulioCesarMatias) [reply src/main/sensors/barometer.c:43]: thanks, the compilation format of EmuFlight, BetaFlight and iNav is still pretty confusing to me

File: make/source.mk [hunk 2]
```diff
@@ -252,6 +252,7 @@ SIZE_OPTIMISED_SRC := $(SIZE_OPTIMISED_SRC) \
             drivers/barometer/barometer_bmp280.c \
             drivers/barometer/barometer_fake.c \
             drivers/barometer/barometer_ms5611.c \
+            drivers/barometer/barometer_dps310.h \
```
Comment (Quick-Flash) [inline make/source.mk:255]: change to barometer_dps310.c instead of .h

File: src/main/target/MATEKF722/target.mk [hunk 3]
```diff
@@ -8,6 +8,7 @@ TARGET_SRC = \
             drivers/barometer/barometer_bmp085.c \
             drivers/barometer/barometer_bmp280.c \
             drivers/barometer/barometer_ms5611.c \
+            drivers/barometer/barometer_dps310.h \
```
Comment (Quick-Flash) [inline src/main/target/MATEKF722/target.mk:11]: change to a .c instead of a .h file.

**Comment by Quick-Flash** [comment]
this needs some testing to verify

**Comment by JulioCesarMatias** [comment]
> this needs some testing to verify

yes, i don't have any card with this barometer. i also couldn't compile to generate the .hex file for @goodnerdboy test and give feedback if it worked or not

**Comment by JulioCesarMatias** [comment]
@Quick-Flash 

https://github.com/emuflight/EmuFlight/blob/master/docs/development/Building%20in%20Windows.md

it was very confusing for me because it was old. I was unable to compile with cygwin, it returns errors

**Comment by goodnerdboy** [comment]
if you give me a .hex i'll be happy to test it

**Comment by nerdCopter** [comment]
> if you give me a .hex i'll be happy to test it
[EmuFlight_0.3.5_DP310-test.zip](https://github.com/emuflight/EmuFlight/files/6542121/EmuFlight_0.3.5_DP310-test.zip)

**Comment by JulioCesarMatias** [comment]
> > if you give me a .hex i'll be happy to test it
> > [EmuFlight_0.3.5_DP310-test.zip](https://github.com/emuflight/EmuFlight/files/6542121/EmuFlight_0.3.5_DP310-test.zip)

Thank you so much for this

**Comment by nerdCopter** [comment]
> @Quick-Flash
> 
> https://github.com/emuflight/EmuFlight/blob/master/docs/development/Building%20in%20Windows.md
> 
> it was very confusing for me because it was old. I was unable to compile with cygwin, it returns errors

it is very easy with Linux or Windows Subsystem for Linux. 
https://github.com/emuflight/EmuFlight/wiki/Building

another good option is using the github actions workflow via tags:
```
git checkout #[master|branch|commit]
git tag buildMe
git push origin buildMe
git tag --delete buildMe   #delete local tag
git push --delete origin buildMe   #delete remote tag
```
then look in your github repo's actions tab. 
for example:
image
image

**Comment by JulioCesarMatias** [comment]
> > @ Quick-Flash
> > https://github.com/emuflight/EmuFlight/blob/master/docs/development/Building%20in%20Windows.md
> > foi muito confuso para mim porque era antigo. Não consegui compilar com o cygwin, ele retorna erros
> 
> é muito fácil com Linux ou subsistema Windows para Linux.
> https://github.com/emuflight/EmuFlight/wiki/Building
> 
> outra boa opção é usar o fluxo de trabalho de ações do github por meio de tags:
> 
> ```
> git checkout #[master|branch|commit]
> git tag buildMe
> git push origin buildMe
> git tag --delete buildMe   #delete local tag
> git push --delete origin buildMe   #delete remote tag
> ```
> 
> em seguida, observe a guia de ações do seu garfo pessoal.
> por exemplo:
> imagem
> imagem

I loved it, thank you!

**Comment by goodnerdboy** [comment]
> > if you give me a .hex i'll be happy to test it
> > [EmuFlight_0.3.5_DP310-test.zip](https://github.com/emuflight/EmuFlight/files/6542121/EmuFlight_0.3.5_DP310-test.zip)

Hi. Tried the MATEK F722 SE target on a F722 SE with a DPS310 with no luck.
The baro is not recognised in auto, and setting it DPS310 in CLI has no other effet than having the baro_hardware set.
status
tasks

**Comment by JulioCesarMatias** [comment]
> Hi. Tried the MATEK F722 SE target on a F722 SE with a DPS310 with no luck.
> The baro is not recognised in auto, and setting it DPS310 in CLI has no other effet than having the baro_hardware set.


in the issue you said you were using the matek f722, but you didn't say it was the SE version, so I didn't put the support on it. I'm going to add PR for the SE version.

**Comment by nerdCopter** [comment]
@JulioCesarMatias , can you add to this branch/PR?  normally i like "atomic" PR's but in this case, i would be good to "fix both" at once.

**Comment by JulioCesarMatias** [comment]
> @JulioCesarMatias , can you add to this branch/PR? normally i like "atomic" PR's but in this case, i would be good to "fix both" at once.

yes, I will add here in this branch

**Comment by nerdCopter** [comment]
[EmuFlight_0.3.5_MATEKF722SE_Build_8ec625ed0_DPS310_test.hex.zip](https://github.com/emuflight/EmuFlight/files/6548011/EmuFlight_0.3.5_MATEKF722SE_Build_8ec625ed0_DPS310_test.hex.zip)

**Comment by goodnerdboy** [comment]
> [EmuFlight_0.3.5_MATEKF722SE_Build_8ec625ed0_DPS310_test.hex.zip](https://github.com/emuflight/EmuFlight/files/6548011/EmuFlight_0.3.5_MATEKF722SE_Build_8ec625ed0_DPS310_test.hex.zip)

Hi again.
Sorry but still no luck with this last build.
No Auto detect, and nothing happens when set the baro_hardware = DSP310 in cli.

status_new_build


And here is what it should be (from betaflight on the same board) : 

status_new_BF

**Comment by goodnerdboy** [comment]
ok got it to work by setting baro_bustype = I2C, baro_i2c_device = 1 and baro_hardware = AUTO

**Comment by JulioCesarMatias** [comment]
> ok got it to work by setting baro_bustype = I2C, baro_i2c_device = 1 and baro_hardware = AUTO
> 
> Still, something is off. The sensor's tab shows a laggy and steppy curve where it should be way smoother. But maybe there is something i don't set correctly.
> 
> From Emu :
> 
> sensor_emu
> tasks_emu
> 
> and from BF :
> 
> sensor_BF
> tasks_BF

with the corrections I just made, I believe that now it will work normal

https://github.com/emuflight/EmuFlight/pull/578/commits/52e273bd5139b6dba48fec392a180dddbf64040b

**Comment by JulioCesarMatias** [comment]
> ok got it to work by setting baro_bustype = I2C, baro_i2c_device = 1 and baro_hardware = AUTO
> 
> Still, something is off. The sensor's tab shows a laggy and steppy curve where it should be way smoother. But maybe there is something i don't set correctly.
> 
> From Emu :
> 
> sensor_emu
> tasks_emu
> 
> and from BF :
> 
> sensor_BF
> tasks_BF

with the corrections I just made, I believe that now it will work normal

https://github.com/emuflight/EmuFlight/pull/578/commits/52e273bd5139b6dba48fec392a180dddbf64040b

**Comment by JulioCesarMatias** [comment]
ok to generate the .hex and run test

**Comment by nerdCopter** [comment]
[MATEK_DPS310_testing_9fa0bc55c.zip](https://github.com/emuflight/EmuFlight/files/6555166/MATEK_DPS310_testing_9fa0bc55c.zip)

**Comment by JulioCesarMatias** [comment]
> [MATEK_DPS310_testing_9fa0bc55c.zip](https://github.com/emuflight/EmuFlight/files/6555166/MATEK_DPS310_testing_9fa0bc55c.zip)


I hadn’t seen it, the HD and PX versions use DPS310 in the new versions
Captura de tela 2021-05-27 132551

**Comment by nerdCopter** [comment]
[MATEK_DPS310_testing_2af0eeaa2.zip](https://github.com/emuflight/EmuFlight/files/6555553/MATEK_DPS310_testing_2af0eeaa2.zip)

**Comment by goodnerdboy** [comment]
so still has to set the bus type and i2c device by hand in the CLI
This works. But as soon as i set the baro_hardware to AUTO or DPS310 the board doesn't boot anymore. The Fc status LED stay off.

**Comment by JulioCesarMatias** [comment]
the i2c itself is killing the machine cycle of the microcontroller, I'm trying to find it, but I don't know what it is

File: src/main/drivers/barometer/barometer_dps310.c [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+        *pressure = baroState.pressure; 
+    }
+
+    if (temperature) {
+        *temperature = (baroState.temperature * 100);   // to centidegrees
+    }
+}
+
+#define DETECTION_MAX_RETRY_COUNT   5
+static bool deviceDetect(busDevice_t * busDev)
```
Comment (JulioCesarMatias) [inline src/main/drivers/barometer/barometer_dps310.c:289]: I believe it is this function that is causing problems

Comment (JulioCesarMatias) [reply src/main/drivers/barometer/barometer_dps310.c:289]: I'm replacing it with another function, and implemented a 10ms delay to start

**Comment by JulioCesarMatias** [comment]
@goodnerdboy laggy and steppy stopped?

new commit https://github.com/emuflight/EmuFlight/pull/578/commits/f185c9a2658538e01f4cff009561deba53e85807

**Comment by nerdCopter** [comment]
[MATEK_DPS310_testing_be7b986a3.zip](https://github.com/emuflight/EmuFlight/files/6561078/MATEK_DPS310_testing_be7b986a3.zip)


Should new and separate target/target-names be created just for the DPS310 versions?

**Comment by goodnerdboy** [comment]
The board now doesn't boot normally after flashing and there is no way to connect to it through the COM

**Comment by nerdCopter** [comment]
@goodnerdboy , hold the boot-button while plugging in the USB.  this will activate DFU mode for re-flashing your preferred .hex

**Comment by goodnerdboy** [comment]
> @goodnerdboy , hold the boot-button while plugging in the USB. this will activate DFU mode for re-flashing your preferred .hex

i know but thanks :)
I was just describing the effect of flashing this last .hex

**Comment by JulioCesarMatias** [comment]
> [MATEK_DPS310_testing_be7b986a3.zip](https://github.com/emuflight/EmuFlight/files/6561078/MATEK_DPS310_testing_be7b986a3.zip)
> 
> Devem ser criados nomes de destino / destino novos e separados apenas para as versões DPS310?
> por exemplo `MATEKF722SE_DPS310`??

@goodnerdboy 
did you try that?

**Comment by goodnerdboy** [comment]
> > [MATEK_DPS310_testing_be7b986a3.zip](https://github.com/emuflight/EmuFlight/files/6561078/MATEK_DPS310_testing_be7b986a3.zip)
> > Devem ser criados nomes de destino / destino novos e separados apenas para as versões DPS310?
> > por exemplo `MATEKF722SE_DPS310`??
> 
> @goodnerdboy
> did you try that?

Yes, it is the one "bricking" the board. Once flashed the board will not reboot correctly and there is no way to connect to it (but dfu)

**Comment by JulioCesarMatias** [comment]
@goodnerdboy I thought it was the other function, so I was wrong. I had substituted the function `static bool deviceConfigure (busDevice_t * busDev)`  for the same one as INAV, when it was equal to BetaFlight it did not cause a reboot. BetaFlight.
I am going to put the BetaFlight function back on.

**Comment by nerdCopter** [comment]
[MATEK_DPS310_testing_01f3143f9.zip](https://github.com/emuflight/EmuFlight/files/6577391/MATEK_DPS310_testing_01f3143f9.zip)

**Comment by JulioCesarMatias** [comment]
@goodnerdboy tested?

**Comment by nerdCopter** [comment]
no testing, no progress?

**Comment by JulioCesarMatias** [comment]
> no testing, no progress?

without tests, I tried some friends here in Brazil with F722, but they all use the BMP280.

**Comment by RatDad1994** [comment]
> > no testing, no progress?
> 
> without tests, I tried some friends here in Brazil with F722, but they all use the BMP280.

I have an iFlight F7 Beast, with the DPS310 baro, I'd do some testing for you if I can.

**Comment by nerdCopter** [comment]
@JulioCesarMatias cc:@RatDad1994  , IFlight F7 Beast (IFRC_IFLIGHT_F745_AIO) is in this pull-request if you like to get the target files:
https://github.com/emuflight/EmuFlight/pull/509/

The full PR 509  is not up to standard for merge, by the way.

**Comment by JulioCesarMatias** [comment]
I spent all afternoon trying to get the DPS310 to work on Emu, and I couldn't. I used a Matek F405-SE for testing. I don't know why Emu can't recognize the DPS310, I will close this PR.

**Comment by nerdCopter** [comment]
any further thoughts on this?

**Comment by nerdCopter** [comment]
re-opening just to attempt merge-master.

**Pull request closed without merging**